bento_beacon/endpoints/info.py

```diff
@@ -114,48 +114,74 @@ def get_experiment_schema():
 # -------------------------------------------------------
 # these return the appropriate response but also save as a side effect
 
-def build_service_info():
-    service_info = deepcopy(current_app.config["BEACON_CONFIG"].get("serviceInfo"))
-    service_info["environment"] = "dev" if current_app.config["DEBUG"] else "prod"
-    service_info["id"] = current_app.config["BEACON_ID"]
-    service_info["name"] = current_app.config["BEACON_NAME"]
+
+def build_service_details():
+    # build info response in beacon format
+    info = current_app.config["BEACON_CONFIG"].get("serviceInfo")
+    s = {
+        "id": current_app.config["BEACON_ID"],
+        "name": current_app.config["BEACON_NAME"],
+        "apiVersion": current_app.config["BENTO_BEACON_VERSION"],
+        "environment": "dev" if current_app.config["DEBUG"] else "prod",
+        "organization": info["organization"]
+    }
+
+    # url for beacon ui
+    if current_app.config["BEACON_UI_ENABLED"]:
+        s["welcomeUrl"] = current_app.config["BEACON_UI_URL"]
 
     # retrieve dataset description from DATS
     # may be multiple datasets, so collect all descriptions into one string
     # for custom description, add a "description" field to service info in beacon_config.json
     k_datasets = katsu_datasets()
     dats_array = list(map(lambda d: json.loads(d.get("datsFile", "{}")), k_datasets))
     description = " ".join([d.get("description") for d in dats_array if "description" in d])
-    if description and service_info.get("description") is None:
-        service_info["description"] = description
+    if description and info.get("description") is None:
+        s["description"] = description
 
-    # url for beacon ui
-    if current_app.config["BEACON_UI_ENABLED"]:
-        service_info["welcomeUrl"] = current_app.config["BEACON_UI_URL"]
-
-    current_app.config["SERVICE_INFO"] = service_info
-    return service_info
+    current_app.config["SERVICE_DETAILS"] = s
+    return s
 
 
 def build_ga4gh_service_info():
-    service_info = current_app.config.get("SERVICE_INFO", build_service_info())
-    service_info["type"] = {
-        "artifact": "Beacon v2",
-        "group": "org.ga4gh",
-        "version": current_app.config["BEACON_SPEC_VERSION"]
+    # construct from beacon-format info
+    info = current_app.config.get("SERVICE_DETAILS", build_service_details())
+    beacon_spec_version = current_app.config["BEACON_SPEC_VERSION"]
+
+    s = {
+        "id": info["id"],
+        "name": info["name"],
+        "type": {
+            "artifact": "Beacon v2",
+             "group": "org.ga4gh",
+             "version": beacon_spec_version
+        },
+        "environment": info["environment"],
+        "organization": {
+            "name": info["organization"]["name"],
+            "url": info["organization"]["welcomeUrl"]
+        },
+        "contactUrl": info["organization"]["contactUrl"],
+        "version": info["apiVersion"],
```

Comment:
this should be the version of the service, not the API

Reply:
spec says "'Version of the API or specification"

https://github.com/ga4gh-discovery/ga4gh-service-info/blob/develop/service-info.yaml#L121

Reply:
that's for service type, not info. see https://github.com/ga4gh-discovery/ga4gh-service-info/blob/develop/service-info.yaml#L99

Reply:
this is what I get for reading specs by hand